packages/ui-causal-graph-editor/src/shared/use-causal-graph-editor.tsx

```diff
@@ -70,6 +77,19 @@ export default function useCausalGraphEditor(
         }, {} as GraphApi);
     }, [dispatch]);
 
+    useEffect(() => {
+        // If we have a time series causasl graph and if the layout chosen supports tiers and these are not defined
+        if (
+            isTimeSeriesCausalGraph &&
+            'tiers' in graphLayout &&
+            'orientation' in graphLayout &&
+            graphLayout.tiers === undefined
+        ) {
+            // We set tiers based on TimeSeriesCausalGraph structure of node lags
+            graphLayout.tiers = { group: 'variable_name', order_nodes_by: 'time_lag' };
```

Comment:
Thinking about it more, putting this logic in an effect is also not great as this is just mutating a prop, not causing a re-render so it will cause inconsistent state (the component will first render with the layout unchanged, then the effect will run and mutate it without causing a re-render).

There's different ways to do it better, one could be to make this a memo (as it runs synchronously rather than after the render) and *return* the mutated layout from this hook. Then in the component you'll use the updated layout instead of the original. 

Let me know if this makes sense or if you have another idea of how we can make this work better, happy to discuss further.

Reply:
I think the memo idea sounds good, updated to follow it and since we now pass graphLayout to the hook I removed the last param since we can get it within.

Reply:
Nice, seems correct now :)

Reply:
Thank you for the help :)